py3status/modules/usbguard.py

```diff
@@ -0,0 +1,358 @@
+
+        self.parent.all_devices = self.parent._get_all_devices()
+        self.parent.py3.update()
+
+    def run(self):
+        while not self.parent.killed.is_set():
+            self.parent._init_dbus()
+            self.parent.dbus.subscribe(
+                object=self.parent.dbus_devices,
+                signal='DevicePresenceChanged',
+                signal_fired=self._on_devices_presence_changed,
+            )
+
+            self.parent.dbus.subscribe(
+                object=self.parent.dbus_devices,
+                signal='DevicePolicyChanged',
+                signal_fired=self._on_devices_policy_changed,
+            )
+
+            self.loop = GLib.MainLoop()
+            self.loop.run()
+
+
+class Py3status:
+    """
+    """
+
+    button_allow = 1
+    button_block = 3
+    button_reject = None
+    format = u'{format_all_devices}'
+    format_all_devices = u'\?color=rule {name} is {rule}'
+    format_device_separator = u'\?color=separator  \| '
+    format_notification = u'{name} is {action}'
+    thresholds = {
+        'action': [('block', "bad"), ('reject', "degraded"), ('allow', "good")],
+        'rule': [('block', "bad"), ('reject', "degraded"), ('allow', "good")],
+    }
+
+    def _init_dbus(self):
+        self.dbus_interface = 'org.usbguard'
+        self.dbus_devices = '/org/usbguard/Devices'
+        self.dbus = SystemBus()
+        self.error = None
+        try:
+            self.proxy = self.dbus.get(self.dbus_interface, self.dbus_devices)
+        except:  # noqa e722
+            self.error = Exception(STRING_USBGUARD_DBUS)
+
+    def _get_all_devices(self):
+        devices_array = []
+        response = {}
+        # TODO: filter, match, allow, block
+        devices = self.proxy.listDevices('match')
+        keys = [
+            'serial',
+            'rule',
+            'id',
+            'name',
+            'hash',
+            'parent_hash',
+            'via_port',
+            'with_interface',
+        ]
+        # _regex_serial = re.compile(r'\S*serial \"(\S+)\"\S*')
+        # _regex_rule = re.compile(r'^(\S+)')
+        # _regex_id = re.compile(r'id (\S+)')
+        # _regex_name = re.compile(r'name \"(.*)\" hash')
+        # _regex_hash = re.compile(r'hash \"(.*)\" parent-hash')
+        # _regex_parent_hash = re.compile(r'parent-hash \"(.*)\" via-port')
+        # _regex_via_port = re.compile(r'via-port \"(.*)\" with-interface')
+        # _regex_with_interface = re.compile(r'with-interface \{ (.*) \}$')
+
+        for usbguard_id, device in devices:
+            params = {}
+            params['usbguard_id'] = usbguard_id
+            for key in keys:
+                value = None
+                regex = eval('_regex_' + key)
+                value = regex.findall(device)
+                if value:
+                    value = value[0]
+                    value = value.encode('latin-1').decode('unicode_escape')
+                    value = value.encode('latin-1').decode('utf-8')
```

Comment:
You have some cipher USB devices?

Reply:
Yes see this:
![image](https://user-images.githubusercontent.com/519083/44806826-40933c00-abc8-11e8-9668-5e00cabc9c76.png)